evalml/automl/utils.py

```diff
@@ -123,7 +132,14 @@ def tune_binary_threshold(
     ):
         pipeline.threshold = 0.5
         if X_threshold_tuning is not None:
-            y_predict_proba = pipeline.predict_proba(X_threshold_tuning)
+            if is_time_series(problem_type):
+                X_threshold_tuning = X_threshold_tuning.iloc[
+                    : pipeline.forecast_horizon
+                ]
+                y_threshold_tuning = y_threshold_tuning.iloc[
+                    : pipeline.forecast_horizon
+                ]
```

Comment:
`predict_proba` is limited to `forecast_horizon` observations

Reply:
I think we can use `predict_proba_in_sample` here?